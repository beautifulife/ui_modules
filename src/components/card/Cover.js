import React from 'react';
import PropTypes from 'prop-types';
import './Cover.scss';
import logo from '../../assets/images/logo.jpg';

const Cover = ({ imgSrc, label, type }) => {
  const styles = {
    background: {
      backgroundImage: `url(${imgSrc || logo})`
    }
  };

  return (
    <div className={`Cover ${type}`}>
      <div
        className="img"
        style={styles.background}
        aria-label={label || 'sub-image'}
      />
    </div>
  );
};

Cover.propTypes = {
  imgSrc: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.string
};

Cover.defaultProps = {
  imgSrc: '',
  label: '',
  type: ''
};

export default Cover;
