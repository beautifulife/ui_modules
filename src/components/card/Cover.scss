.Cover {
  overflow: hidden;

  &.vertical {
    .img {
      width: 100%;
      border-radius: 3px 3px 0 0;
      background-size: cover;
      background-position: center center;

      &::after {
        content: '';
        display: block;
        padding-bottom: 100%;
      }
    }
  }

  &.horizontal {
    width: 150px;
    margin-right: 5px;

    .img {
      height: 100%;
      border-radius: 3px 0 0 3px;
      background-size: cover;
      background-position: center center;

      &::after {
        content: '';
        display: block;
        padding-bottom: 100%;
      }
    }
  }
}
