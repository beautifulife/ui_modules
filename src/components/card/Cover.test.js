import React from 'react';
import { shallow } from 'enzyme';
import Cover from './Cover';

describe('Cover', () => {
  describe('about render', () => {
    const wrapper = shallow(<Cover />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
