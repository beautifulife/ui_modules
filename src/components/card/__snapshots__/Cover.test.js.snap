// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cover
    imgSrc=""
    label=""
    type=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        aria-label="sub-image"
        className="img"
        style={
          Object {
            "backgroundImage": "url(logo.jpg)",
          }
        }
      />,
      "className": "Cover ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "sub-image",
        "className": "img",
        "style": Object {
          "backgroundImage": "url(logo.jpg)",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          aria-label="sub-image"
          className="img"
          style={
            Object {
              "backgroundImage": "url(logo.jpg)",
            }
          }
        />,
        "className": "Cover ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "sub-image",
          "className": "img",
          "style": Object {
            "backgroundImage": "url(logo.jpg)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
